library/src/main/java/com/google/android/exoplayer2/text/ssa/Style.java

```diff
@@ -0,0 +1,283 @@
+package com.google.android.exoplayer2.text.ssa;
+
+import java.util.Map;
+
+import static android.R.attr.angle;
+
+/**
+ * Created by cablej01 on 27/12/2016.
+ */
+
+public class Style {
```

Comment:
The way we typically support styling of subtitles is by attaching Span objects directly onto the Cue text, which needs to be a CharSequence (you can use SpannableStringBuilder directly) rather than just a String. Take a look at what we do for webvtt to get the idea (see in particular WebvttCueParser).

I think you should use the same approach for SSA. In particular, there are built in Span objects for font, background color, foreground color, size adjustment, bold, italic, underline, strike-out etc, which the existing subtitle rendering components will render automatically without you having to do anything.

For anything that's not supported by existing Span objects I guess it is sensible to extend Cue to add them (assuming they apply to the whole Cue). If they might apply to only part of the Cue, a custom Span would be appropriate. This stuff is only useful if you think someone's going to build a UI component that actually uses this information.